### PR TITLE
drivers:dac:ltc2688: Minor Update

### PR DESCRIPTION
**Pull request by Pratyush-Mallick**
Added an extra parameter in device descriptor to store the computed dac code.

Signed-off-by: Pratyush Mallick <Pratyush.Mallick@analog.com>

### CODE REVIEW

**Review by amiclaus — APPROVED** [review]

**Review by adrimbarean — APPROVED** [review]

File: drivers/dac/ltc2688/ltc2688.c [hunk 1]
```diff
@@ -387,6 +387,8 @@ int32_t ltc2688_set_voltage(struct ltc2688_dev *dev, uint8_t channel,
 	if(code > 0xFFFF)
 		code = 0xFFFF;
 
+	dev->dac_code[channel] = code;
```
Comment (buha) [inline drivers/dac/ltc2688/ltc2688.c:390]: what is the purpose of this change ? I expect the code to make it somehow to the registers of the chip but the code now simply stores it...

Comment (pmallick-adi) [reply drivers/dac/ltc2688/ltc2688.c:390]: To let the user know of what the actual code that is being written into the device register after taking gain and offset into consideration.

Comment (Pratyush-Mallick) [reply drivers/dac/ltc2688/ltc2688.c:398]: To let the user know of what the actual code that is being written into the device register after taking gain and offset into consideration.

**Comment by Pratyush-Mallick** [comment]
Changelog:
Storing the dac power settings in the ltc2688_set_pwr_dac() function.

**Comment by pmallick-adi** [comment]
Do the changes look good to you?

File: drivers/dac/ltc2688/ltc2688.c (hunk 1 above)
Comment (buha) [reply drivers/dac/ltc2688/ltc2688.c:398]: It seems to me then that this is a debug feature. I would advise against implementing it this way and suggest using `pr_debug` from this file to achieve what you want.
https://github.com/analogdevicesinc/no-OS/blob/master/include/no_os_print_log.h

This computed code is information that the driver doesn't need to store in the device descriptor (proof: it's not used elsewhere than the function it's computed in: ltc2688_set_voltage).

File: drivers/dac/ltc2688/ltc2688.c [hunk 2]
```diff
@@ -140,8 +140,16 @@ static int32_t _ltc2688_spi_update_bits(struct ltc2688_dev *dev, uint8_t reg,
  */
 int32_t ltc2688_set_pwr_dac(struct ltc2688_dev *dev, uint16_t setting)
 {
-	return _ltc2688_spi_write(dev, LTC2688_CMD_POWERDOWN_REG,
-				  setting);
+	int32_t ret;
```
Comment (buha) [inline drivers/dac/ltc2688/ltc2688.c:143]: this should be in a separate commit, it seems unrelated to the dac code change

File: drivers/dac/ltc2688/ltc2688.c (hunk 1 above)
Comment (pmallick-adi) [reply drivers/dac/ltc2688/ltc2688.c:398]: It's not about just printing the value, it about getting the dac codes back to application level. I believe the device descriptor should provide information about what's the current configuration of DAC registers. In this [drivers ](https://github.com/analogdevicesinc/no-OS/blob/master/drivers/dac/ad5770r/ad5770r.h#L317) as well, the dac value is being stored in the descriptor itself and it is not being used anywhere in the drivers except for setting the dac voltage

If _ltc2688_spi_read() function were not declared static, it would have been easier to read contents of the dac registers directly.
Or else I can create a global variable to store these values in the drivers itself.

File: drivers/dac/ltc2688/ltc2688.c (hunk 2 above)
Comment (pmallick-adi) [reply drivers/dac/ltc2688/ltc2688.c:143]: Will add a separate commit for this.

File: drivers/dac/ltc2688/ltc2688.c (hunk 1 above)
Comment (Pratyush-Mallick) [reply drivers/dac/ltc2688/ltc2688.c:398]: It's not about just printing the value; it's about getting the DAC codes back to the application level. I believe the device descriptor should provide information about the current configuration of DAC registers. In this [drivers ](https://github.com/analogdevicesinc/no-OS/blob/master/drivers/dac/ad5770r/ad5770r.h#L317) as well, the DAC value is being stored in the descriptor itself, and it is not being used anywhere in the drivers except for setting the DAC voltage 

If the _ltc2688_spi_read() function were not declared static, it would have been easier to read the contents of the DAC registers directly. Or else I can create a global variable to store these values in the drivers themselves.

File: drivers/dac/ltc2688/ltc2688.c (hunk 2 above)
Comment (Pratyush-Mallick) [reply drivers/dac/ltc2688/ltc2688.c:143]: Will do it.

File: drivers/dac/ltc2688/ltc2688.c (hunk 1 above)
Comment (buha) [reply drivers/dac/ltc2688/ltc2688.c:398]: Ok, fine for me then if you say this should work as a cached version of the value one would read using _ltc2688_spi_read()

**Review by buha — APPROVED** [review]
looks good, waiting for splitting in two commits and will merge then

**Comment by pmallick-adi** [comment]
Can you please merge this ASAP. There is an urgent requirement for these drivers.
